src/ray/gcs/gcs_server/gcs_init_data.cc

```diff
@@ -14,18 +14,17 @@
 
 #include "ray/gcs/gcs_server/gcs_init_data.h"
 
+#include <boost/thread/latch.hpp>
+
 namespace ray {
 namespace gcs {
-void GcsInitData::AsyncLoad(const EmptyCallback &on_done) {
+
+void GcsInitData::Load() {
   // There are 5 kinds of table data need to be loaded.
-  auto count_down = std::make_shared<int>(5);
-  auto on_load_finished = [count_down, on_done] {
-    if (--(*count_down) == 0) {
-      if (on_done) {
-        on_done();
-      }
-    }
-  };
+  boost::latch latch(5);
+  latch.count_down();
```

Comment:
Why do you count down right after creation?

Reply:
fixed